### PR TITLE
added ci workflow for linting and formatted the code with black

### PR DESCRIPTION
**Pull request by 0xcha05**

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        with:
+          python-version: "3.11"
+
+      - name: Install Black
+        run: |
+          python -m pip install --upgrade pip
+          pip install click==7.1.2
+          pip install black==21.4b2
+
+      - name: Run Black
```
Comment (astelmach01) [inline .github/workflows/lint.yml:27]: Personally, I would avoid actually modifying code in the CI. Instead, flake8 should be used as a linter. 

I would go with pre-commit hooks instead to ensure committed code is formatted.

[My pre commit hooks](https://github.com/astelmach01/Cosmo-backend/blob/main/.pre-commit-config.yaml) is more than enough.

**Review by astelmach01 — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yml (hunk 1 above)
Comment (0xcha05) [reply .github/workflows/lint.yml:27]: it doesn't modify the code, just does a check.

Comment (astelmach01) [reply .github/workflows/lint.yml:27]: Ah you're right, I would still add pre commit hooks nonetheless

Comment (gowtham-ryze) [reply .github/workflows/lint.yml:27]: This is widely used and a common practice. if you want to have both this and pre-commit hooks, we can add that in a separate PR.

Comment (0xcha05) [reply .github/workflows/lint.yml:27]: it's a pretty common practice, if you want both this and pre-commit hooks, we can do it in a different PR.

**Pull request closed without merging**